revolv/donor/views.py

```diff
@@ -30,3 +34,78 @@ def get_context_data(self, **kwargs):
         context["donor_has_no_donated_projects"] = Project.objects.donated_projects(self.user_profile).count() == 0
 
         return context
+
+class AccountingView(UserDataMixin, TemplateView):
+    """
+    View of the accounting page.
+    """
+    template_name = 'donor/accounting.html'
+
+    def get_context_data(self, **kwargs):
+        context = super(AccountingView, self).get_context_data(**kwargs)
+
+        context['project_names'] = [p.title for p in Project.objects.all()]
+        context['admin_repayments'] = AdminRepayment.objects.repayments()
+        context['admin_reinvestments'] = AdminReinvestment.objects.reinvestments()
+        context['donations'] = Payment.objects.donations()
+        context['jsonurl'] = unicode(str(reverse('donor:accountingPartial')))
+
+        context['clean_project_names'] = {}
+        clean_project_names = context['clean_project_names']
+        for p in context['project_names']:
+            clean_name = AccountingAggregator().clean_project_name(p)
+            clean_project_names[clean_name] = p
+
+        return context
+
+class AccountingPartialView(TemplateView):
+    """
+    A view that returns partial HTML of all relevant accounting data.
+    This is queried by an AJAX request when creating the Accounting page.
```

Comment:
You should say that not only is it queried when creating the accounting page, but when requesting data for new dates as well.